frontend/src/metabase/containers/SaveQuestionModal.tsx

```diff
@@ -158,10 +160,10 @@ export const SaveQuestionModal = ({
       ? t`First, save your question`
       : t`First, save your model`;
 
+  const isSavedQuestionChanged = useSelector(getIsSavedQuestionChanged);
   const showSaveType =
-    !question.isSaved() &&
-    !!originalQuestion &&
-    !originalQuestion.isDataset() &&
+    isSavedQuestionChanged &&
+    originalQuestion != null &&
```

Comment:
Even though, `originalQuestion` would be guaranteed to have value when `isSavedQuestionChanged` is true, TypeScript won't know that, so I need to still put `originalQuestion != null` here to stop TypeScript from complaning that `originalQuestion` could be null.